site/docs/plugins/commands.md

```diff
@@ -5,10 +5,626 @@ next: false
+
+Here's an example of a scoped command:
+
+```js
+const myCommands = new CommandGroup();
+
+myCommands
+  .command("start", "Initializes bot configuration")
+  .addToScope(
+    { type: "all_private_chats" },
+    (ctx) => ctx.reply(`Hello, ${ctx.chat.first_name}!`),
+  )
+  .addToScope(
+    { type: "all_group_chats" },
+    (ctx) => ctx.reply(`Hello, members of ${ctx.chat.title}!`),
+  );
+```
+
+The `start` command can now be called from both private and group chats, and it will give a different response depending on where it gets called from.
+Now if you call `myCommands.setCommands`, the `start` command will be registered to both private and group chats.
+
+Here's an example of a command that's only accessible to group admins.
+
+```js
+adminCommands
+  .command("secret", "Admin only")
+  .addToScope(
+    { type: "all_chat_administrators" },
+    (ctx) => ctx.reply("Free cake!"),
+  );
+```
+
+And here is an example of a command that's only accessible in groups
+
+```js
+myCommands
+  .command("fun", "Laugh")
+  .addToScope(
+    { type: "all_group_chats" },
+    (ctx) => ctx.reply("Haha"),
+  );
+```
+
+Notice that when you call the `command` method, it opens up a new command.
+If you give it a handler, that handler will apply to the `default` scope of that command.
+Calling `addToScope` on that command will then add a new handler, which will be filtered to that scope.
+Take a look at this example.
+
+```ts
+myCommands
+  .command(
+    "default",
+    "Default command",
+    // This will be called when not on a group chat, or when the user is not an admin
+    (ctx) => ctx.reply("Hello from default scope"),
+  )
+  .addToScope(
+    { type: "all_group_chats" },
+    // This will only be called for non-admin users in a group
+    (ctx) => ctx.reply("Hello, group chat!"),
+  )
+  .addToScope(
+    { type: "all_chat_administrators" },
+    // This will be called for group admins, when inside that group
+    (ctx) => ctx.reply("Hello, admin!"),
+  );
+```
+
+## Command Translations
+
+Another powerful feature is the ability to set different names for the same command, and their respective descriptions based on the user language.
+The commands plugin makes that easy by providing the `localize` method.
+Check it out:
+
+```js
+myCommands
+  // You need to set a default name and description
+  .command("hello", "Say hello")
+  // And then you can set the localized ones
+  .localize("pt", "ola", "Dizer olá");
+```
+
+Add as many as you want! The plugin will take care of registering them for you when you call `myCommands.setCommands`.
+
+For convenience grammY exports a `LanguageCodes` enum-like object, that you can use for a more idiomatic approach:
+
+::: code-group
+
+```ts [TypeScript]
+import { LanguageCodes } from "grammy/types";
+
+myCommands.command(
+  "chef",
+  "Steak delivery",
+  async (ctx) => await ctx.reply("Steak on the plate!"),
```

Comment:
```suggestion
  (ctx) => ctx.reply("Steak on the plate!"),
```

Reply:
(because we either have to use the concise version here or add `async`/`await` everywhere, but it's odd to be inconsistent about it)